From: Lennart Poettering <lennart@poettering.net>
Date: Fri, 13 Jun 2014 19:05:22 +0200
Subject: rpm: don't hardcode the binary paths in the macros, rely on $PATH

this gives us a little bit more freedom to move things around later on,
as we don't hardcode the systemd paths in old RPMs that shall work with
new systemds.

(cherry picked from commit 8cf7c96517c3688d29b30651adcda9e71e7fcb07)
(cherry picked from commit 2ea8cba266174f4c4afe7b0a5fbbfebf93fb37cc)
---
 src/core/macros.systemd.in | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/src/core/macros.systemd.in b/src/core/macros.systemd.in
index 89b4825..167016a 100644
--- a/src/core/macros.systemd.in
+++ b/src/core/macros.systemd.in
@@ -37,42 +37,42 @@ Requires(postun): systemd \
 %systemd_post() \
 if [ $1 -eq 1 ] ; then \
         # Initial installation \
-        @rootbindir@/systemctl preset %{?*} >/dev/null 2>&1 || : \
+        systemctl preset %{?*} >/dev/null 2>&1 || : \
 fi \
 %{nil}
 
 %systemd_preun() \
 if [ $1 -eq 0 ] ; then \
         # Package removal, not upgrade \
-        @rootbindir@/systemctl --no-reload disable %{?*} > /dev/null 2>&1 || : \
-        @rootbindir@/systemctl stop %{?*} > /dev/null 2>&1 || : \
+        systemctl --no-reload disable %{?*} > /dev/null 2>&1 || : \
+        systemctl stop %{?*} > /dev/null 2>&1 || : \
 fi \
 %{nil}
 
 %systemd_postun() \
-@rootbindir@/systemctl daemon-reload >/dev/null 2>&1 || : \
+systemctl daemon-reload >/dev/null 2>&1 || : \
 %{nil}
 
 %systemd_postun_with_restart() \
-@rootbindir@/systemctl daemon-reload >/dev/null 2>&1 || : \
+systemctl daemon-reload >/dev/null 2>&1 || : \
 if [ $1 -ge 1 ] ; then \
         # Package upgrade, not uninstall \
-        @rootbindir@/systemctl try-restart %{?*} >/dev/null 2>&1 || : \
+        systemctl try-restart %{?*} >/dev/null 2>&1 || : \
 fi \
 %{nil}
 
 %udev_hwdb_update() \
-@rootbindir@/udevadm hwdb --update >/dev/null 2>&1 || : \
+udevadm hwdb --update >/dev/null 2>&1 || : \
 %{nil}
 
 %udev_rules_update() \
-@rootbindir@/udevadm control --reload >/dev/null 2>&1 || : \
+udevadm control --reload >/dev/null 2>&1 || : \
 %{nil}
 
 %journal_catalog_update() \
-@rootbindir@/journalctl --update-catalog >/dev/null 2>&1 || : \
+journalctl --update-catalog >/dev/null 2>&1 || : \
 %{nil}
 
 %tmpfiles_create() \
-@rootbindir@/systemd-tmpfiles --create %{?*} >/dev/null 2>&1 || : \
+systemd-tmpfiles --create %{?*} >/dev/null 2>&1 || : \
 %{nil}
